### ISSUE TITLE
Stat difference between Osty sim and this one.

### ISSUE THREAD

**Issue by Broskimage** [wowsims/tbc#238]
**Describe the bug**
I just tried to fiddle a bit with the sim to see if the results are roughly the same as the mage sim we've been using because I do find it interesting that we might be able to sim full raid scenario. I simmed in both simulators with the same parameters and I got a difference in DPS, couldn't really find the difference until I noticed the stats. In this sim I have to manually add 83crit rating, 36sp and substract 11spirit and 1 int.

While I'm writing this the SP seems to be fixed and the crit difference is lowered to 69 crit rating

**To Reproduce**
I'll add the osty mage config file of my sim and the sharable link. Import the osty config here https://cheesehyvel.github.io/magesim-tbc2/ and use the github link for the other.

**Expected behavior**
I kinda expected that maybe one sim did add the average crit from CC to overall crit % while the other did not to lead to a stat difference. With that said though the results should be the same even if that's the case. 

**Sim Links and Screenshots**
https://wowsims.github.io/tbc/mage/index.html#eJy1kT9oFEEUxvPN7B5zo/EeawyTM8W6hZ7CwV7CgWCRwUJioZyVNuqlUFJYhFRqIccVeiaCQqJgsLgyRYqrRBGThVSCVQori4SkUsldRASNOZ25zR6CkkLwMX8ev++9t2/nScdnOeZlBHIYRQllVgMmkI1SVaRKN8ZuXZsU+4hnlyGJ2h+h7m0gSC1Ms7fbkA4tfbC8+gnq3ZrhBtaakIIetKHaVSYP0heTs/wKQdpoZn3fhOylJ1tQdxsIuK0iaMfEtN5D7qf5rbiK5b208hlq5fVuGFFzG2pthnUDBD00n6k1rPdtI/YcWt+xZ6Vl6dcfUHVDewxpriIg+5OXUGF8HHXgJjuxip49rbUU30ejveP+0fQF/V/q/ma30wLE/FmZw+lzst/rIz4w99Tay5EjsfNmxMsLRq7v5njIT0K7o7zEyxhnE26FuTXGHzG3zvg8Q4PxiCHrCChM4dhQMQzNHg5DswvGMedQMZ8Pi8NFywqWh4U5mHFkJBfTVzyRNNYBU12Q0x3wYiwBZ2Pw/PIuiK5qqcwgny3AoySnc1XKsWJGnyiDuptD0hGzj1lS9pSWWXlA3O/3fyaGRBuMtcU7f9HOx+3MJA1HZ7Q8LDNicx1/RkcXdXA8HTd3XQ90n/lQzPoWtTINmcGYZ/TxCw7tsEo=

Osthy sim config:

eyJlcXVpcHBlZCI6eyJ3ZWFwb24iOjMyMzc0LCJvZmZfaGFuZCI6bnVsbCwicmFuZ2VkIjoyODc4MywiaGVhZCI6MzAyMDYsIm5lY2siOjMwMDE1LCJzaG91bGRlciI6MzAyMTAsImJhY2siOjMyNTI0LCJjaGVzdCI6MzAxOTYsIndyaXN0IjozMDg3MCwiaGFuZHMiOjMwMjA1LCJ3YWlzdCI6MzA4ODgsImxlZ3MiOjMxMDU4LCJmZWV0IjozMjIzOSwiZmluZ2VyMSI6MjkzMDUsImZpbmdlcjIiOjMyNTI3LCJ0cmlua2V0MSI6MzI0ODMsInRyaW5rZXQyIjozMDcyMH0sImVuY2hhbnRzIjp7IndlYXBvbiI6Mjc5ODEsIm9mZl9oYW5kIjpudWxsLCJyYW5nZWQiOm51bGwsImhlYWQiOjM1NDQ3LCJuZWNrIjpudWxsLCJzaG91bGRlciI6MzU0MDYsImJhY2siOm51bGwsImNoZXN0IjoyNzk2MCwid3Jpc3QiOjI3OTE3LCJoYW5kcyI6MzM5OTcsIndhaXN0IjpudWxsLCJsZWdzIjozMTM3MiwiZmVldCI6MzQwMDgsImZpbmdlcjEiOjI3OTI0LCJmaW5nZXIyIjoyNzkyNCwidHJpbmtldDEiOm51bGwsInRyaW5rZXQyIjpudWxsfSwiZ2VtcyI6eyJ3ZWFwb24iOltudWxsLG51bGwsbnVsbF0sIm9mZl9oYW5kIjpbbnVsbCxudWxsLG51bGxdLCJyYW5nZWQiOltudWxsLG51bGwsbnVsbF0sImhlYWQiOlszNDIyMCwzMjIwNCxudWxsXSwibmVjayI6W251bGwsbnVsbCxudWxsXSwic2hvdWxkZXIiOlszMjIwNCwzMDU4NixudWxsXSwiYmFjayI6W251bGwsbnVsbCxudWxsXSwiY2hlc3QiOlszMjIwNCwzMjIwNCwzMDYwMF0sIndyaXN0IjpbMzIyMDQsbnVsbCxudWxsXSwiaGFuZHMiOltudWxsLG51bGwsbnVsbF0sIndhaXN0IjpbMzIyMDQsMzIyMDQsbnVsbF0sImxlZ3MiOlszMjIwNCxudWxsLG51bGxdLCJmZWV0IjpbMzIyMDQsMzIyMDQsbnVsbF0sImZpbmdlcjEiOltudWxsLG51bGwsbnVsbF0sImZpbmdlcjIiOltudWxsLG51bGwsbnVsbF0sInRyaW5rZXQxIjpbbnVsbCxudWxsLG51bGxdLCJ0cmlua2V0MiI6W251bGwsbnVsbCxudWxsXX0sImNvbmZpZyI6eyJpdGVyYXRpb25zIjoxMDAwMDAsInJhY2UiOjUsImR1cmF0aW9uIjoxODAsImR1cmF0aW9uX3ZhcmlhbmNlIjoxMCwicm5nX3NlZWQiOjAsImdjZF91bmxvY2tlZCI6ZmFsc2UsImF2Z19zcGVsbF9kbWciOmZhbHNlLCJhZGRpdGlvbmFsX2RhdGEiOmZhbHNlLCJ0YXJnZXRzIjoxLCJtaXNlcnkiOnRydWUsImN1cnNlX29mX2VsZW1lbnRzIjp0cnVlLCJtYWxlZGljdGlvbiI6ZmFsc2UsImp1ZGdlbWVudF9vZl90aGVfY3J1c2FkZXIiOnRydWUsImp1ZGdlbWVudF9vZl93aXNkb20iOnRydWUsInZhbXBpcmljX3RvdWNoIjp0cnVlLCJ2YW1waXJpY190b3VjaF9yZWdlbiI6NjUsInRvdGVtX29mX3dyYXRoIjpmYWxzZSwid3JhdGhfb2ZfYWlyIjp0cnVlLCJtYW5hX3NwcmluZyI6dHJ1ZSwiaW1wcm92ZWRfbWFuYV9zcHJpbmciOnRydWUsImFyY2FuZV9pbnRlbGxlY3QiOnRydWUsImRpdmluZV9zcGlyaXQiOnRydWUsImltcHJvdmVkX2RpdmluZV9zcGlyaXQiOnRydWUsImJsZXNzaW5nX29mX2tpbmdzIjp0cnVlLCJibGVzc2luZ19vZl93aXNkb20iOnRydWUsIm1hcmtfb2ZfdGhlX3dpbGQiOnRydWUsIm1vb25raW5fYXVyYSI6ZmFsc2UsIm1hZ2VfYXJtb3IiOnRydWUsIm1vbHRlbl9hcm1vciI6ZmFsc2UsImluc3BpcmluZ19wcmVzZW5jZSI6ZmFsc2UsImZpcmVfdnVsbmVyYWJpbGl0eSI6ZmFsc2UsIm1haW50YWluX2ZpcmVfdnVsbmVyYWJpbGl0eSI6ZmFsc2UsIndpbnRlcnNfY2hpbGwiOmZhbHNlLCJpbXBfc2FuY3RpdHkiOmZhbHNlLCJmZXJvY2lvdXNfaW5zcGlyYXRpb24iOjAsImZvb2QiOjI3NjU3LCJmbGFzayI6MCwiYmF0dGxlX2VsaXhpciI6MjgxMDMsImd1YXJkaWFuX2VsaXhpciI6MzIwNjcsIndlYXBvbl9vaWwiOjIwNzQ5LCJkcnVtcyI6Mjk1MjksImRydW1zX2ZyaWVuZCI6dHJ1ZSwicG90aW9uIjoyMjgzMiwiY29uanVyZWQiOjIyMDQ0LCJhdGllc2hfbWFnZSI6ZmFsc2UsImF0aWVzaF93YXJsb2NrIjpmYWxzZSwiZXllX29mX3RoZV9uaWdodCI6dHJ1ZSwiY2hhaW5fb2ZfdGhlX3R3aWxpZ2h0X293bCI6dHJ1ZSwiamFkZV9wZW5kYW50X29mX2JsYXN0aW5nIjpmYWxzZSwic2Nyb2xsX29mX3NwaXJpdCI6ZmFsc2UsImtyZWVncyI6dHJ1ZSwiYmxvb2R0aGlzdGxlIjpmYWxzZSwidGlyaXNmYWxfMnNldCI6dHJ1ZSwidGlyaXNmYWxfNHNldCI6dHJ1ZSwidGVtcGVzdF8yc2V0IjpmYWxzZSwidGVtcGVzdF80c2V0IjpmYWxzZSwic3BlbGxmaXJlX3NldCI6ZmFsc2UsInNwZWxsc3RyaWtlX3NldCI6ZmFsc2UsImV0ZXJuYWxfc2FnZSI6dHJ1ZSwid3JhdGhfb2ZfY2VuYXJpdXMiOmZhbHNlLCJibGFkZV9vZl93aXphcmRyeSI6ZmFsc2UsInJvYmVfZWxkZXJfc2NyaWJlcyI6ZmFsc2UsImJsYWRlX29mX2V0ZXJuYWxfZGFya25lc3MiOmZhbHNlLCJtYW5hX2V0Y2hlZF80c2V0IjpmYWxzZSwibWV0YV9nZW0iOjM0MjIwLCJ0cmlua2V0MSI6MzI0ODMsInRyaW5rZXQyIjozMDcyMCwiaW5uZXJ2YXRlIjoxLCJtYW5hX3RpZGUiOnRydWUsImJsb29kbHVzdCI6dHJ1ZSwicG93ZXJfaW5mdXNpb24iOmZhbHNlLCJzeW1ib2xfb2ZfaG9wZSI6ZmFsc2UsIm1haW5fcm90YXRpb24iOjAsInJlZ2VuX3JvdGF0aW9uIjowLCJyZWdlbl9tYW5hX2F0IjoxMCwicmVnZW5fc3RvcF9hdCI6MjAsInJlZ2VuX2FiX2NvdW50IjozLCJhYl9oYXN0ZV9zdG9wIjowLCJmaXJlX2JsYXN0X3dlYXZlIjpmYWxzZSwiY2NfYW1fcXVldWUiOmZhbHNlLCJjY19hbV9yZXBlYXQiOmZhbHNlLCJ0cmlua2V0MV90IjpbNywxMjcsbnVsbCxudWxsXSwidHJpbmtldDJfdCI6WzcsMCxudWxsLG51bGxdLCJhcmNhbmVfcG93ZXJfdCI6WzEyLG51bGwsbnVsbCxudWxsXSwicHJlc2VuY2Vfb2ZfbWluZF90IjpbMCxudWxsLG51bGwsbnVsbF0sImljeV92ZWluc190IjpbNTAsbnVsbCxudWxsLG51bGxdLCJjb2xkX3NuYXBfdCI6WzEyNyxudWxsLG51bGwsbnVsbF0sImNvbWJ1c3Rpb25fdCI6WzEsbnVsbCxudWxsLG51bGxdLCJiZXJzZXJraW5nX3QiOls0MSxudWxsLG51bGwsbnVsbF0sIm1hbmFfdGlkZV90IjpbNjAsbnVsbCxudWxsLG51bGxdLCJibG9vZGx1c3RfdCI6WzcsbnVsbCxudWxsLG51bGxdLCJwb3dlcl9pbmZ1c2lvbl90IjpbMSxudWxsLG51bGwsbnVsbF0sImRydW1zX3QiOlsyNywiIixudWxsLG51bGxdLCJpbm5lcnZhdGVfdCI6Wzk1LG51bGwsbnVsbCxudWxsXSwicG90aW9uX3QiOlsxLDAsbnVsbCxudWxsXSwiY29uanVyZWRfdCI6WzEyLDEzMixudWxsLG51bGxdLCJzeW1ib2xfb2ZfaG9wZV9hdCI6MCwiZXZvY2F0aW9uX2F0Ijo0NS41LCJldm9fdGlja3MiOjAsInRhbGVudHMiOiJodHRwczovL3RiYy53b3doZWFkLmNvbS90YWxlbnQtY2FsYy9tYWdlLzI1MDAyNTAzMDAwMzAxNTAzMzAxMjUtLTA1MzUwMDAzMTAwMzAwMSIsInN0YXRzIjp7ImludGVsbGVjdCI6NjMxLCJzcGlyaXQiOjMxNywibXA1IjowLCJjcml0IjoxNS45OTE1MjE3MzkxMzA0MzYsImhpdCI6NS45NDI5NDc3MDIwNjAyMjMsImhhc3RlIjo3LjQ4MjU2MTgyNjI1MjM3OCwic3BlbGxfcG93ZXIiOjEwMDIsInNwZWxsX3Bvd2VyX2FyY2FuZSI6NTAsInNwZWxsX3Bvd2VyX2Zyb3N0IjowLCJzcGVsbF9wb3dlcl9maXJlIjo1MH0sInRvb2x0aXBzIjpmYWxzZX19

**Comment by jimmyt857** [comment]
Thanks for posting about this! It took a few minutes but I figured out the stat difference is actually just a difference in what's shown: we don't show the bonus crit from Improved Judgement of the Crusader (because its a debuff) but the osty sim does.

The DPS difference you found is real though so I'm going to spend some more time and try to figure this out.

**Comment by jimmyt857** [comment]
Found an actual bug thanks to this, we were not accounting for the mana reduction from Elemental Precision. The rest of the dps gap seems to be caused by very subtle differences in rotation logic; I made a few tweaks and its closer now, but we're still behind by ~30 dps.

Going to leave this comment open as a reminder so I can come back to this later. For now the difference is small enough and consistent enough that it shouldn't affect any gearing decisions, and we still have a lot of sims to write.
